### DISCUSSION TITLE
HYMAP/HYMAP2 standalone run

### DISCUSSION THREAD

**Discussion by Arijit22IITDelhi** [NASA-LIS/LISF#1468] [Q&A] [answered]
Hello, I wanted to ask that is it possible to run HYMAP/HYMAP2 standalone. For example I have already run LSM within LIS and having the runoff outputs. Can I run HYMAP by giving the runoffs from the already generated LSM outputs, without running LSM?

**Comment by agetiran** [comment, marked as answer]
Hi @Arijit22IITDelhi
Yes, it is possible to run HyMAP in offline mode using LIS outputs from a previous run. Those output files must contain surface runoff (Qs) and baseflow (Qsb) fields. There are some changes you need to make in the lis.config file to enable HyMAP in offline mode, and below is the content of a file that I used. Basically, you need to define the LSM as none, as well as the met forcing (the latter is useful to save time not reading forcings that won’t be used). You also need to define the TEMPLATE model time step and some options defining the external runoff data source. Below are the options in the lis.config file that must be changed for HyMAP runs in offline mode.
 
Lis.config file content:

Land surface model:                 none 
Met forcing sources:                  none  
TEMPLATE model timestep:                      "30mn"

External runoff data source:   "LIS runoff output"
LIS runoff output directory:   ENTER_EXISTING_LIS_OUTPUT_DIRECTORY
LIS runoff output interval:    24hr
LIS runoff output domain file: ./is_input.nc 


#Overall driver options:
Running mode:                       "retrospective"
Map projection of the LIS domain:   "latlon"
Number of nests:                    1
Number of surface model types:      1
Surface model types:                "LSM"
Surface model output interval:      1da         # "1da", "1hr"
Land surface model:                 none # "Noah-MP.4.0.1"
Number of met forcing sources:      1
Blending method for forcings:       "overlay"
Met forcing sources:                  none  "MERRA2" "ERA5"
Met forcing chosen ensemble member:     1
Topographic correction method (met forcing):  "none" "lapse-rate and slope-aspect"
Enable spatial downscaling of precipitation:  0 0
Spatial upscaling method (met forcing):       none none
Spatial interpolation method (met forcing):   bilinear bilinear
Temporal interpolation method (met forcing):  linear linear
Enable new zterp correction (met forcing):   1 1
 
#Runtime options:
Forcing variables list file:              ./NoahMP_InputFiles/forcing_variables_v2.txt
Output methodology:                        "2d gridspace"
Output model restart files:                1
Output data format:                        "netcdf"
Output naming style:                       "3 level hierarchy"
Start mode:                             restart 
Starting year:                          2007
Starting month:                         12
Starting day:                           31
Starting hour:                             23
Starting minute:                           30
Starting second:                           0
Ending year:                            2008
Ending month:                           07
Ending day:                             01   
Ending hour:                               0
Ending minute:                             0
Ending second:                             0
Undefined value:                          -9999
Output directory:                         "./OUTPUT/BIF_SLR_100_65/"
Diagnostic output file:                   "./OUTPUT/BIF_SLR_100_65/lislog_files/lislog"
Number of ensembles per tile:       1
 
#The following options are used for subgrid tiling based on vegetation
Maximum number of surface type tiles per grid:     1
Minimum cutoff percentage (surface type tiles):    0.05
Maximum number of soil texture tiles per grid:     1
Minimum cutoff percentage (soil texture tiles):    0.05
Maximum number of soil fraction tiles per grid:    1
Minimum cutoff percentage (soil fraction tiles):   0.05
Maximum number of elevation bands per grid:        1
Minimum cutoff percentage (elevation bands):       0.05
Maximum number of slope bands per grid:            1
Minimum cutoff percentage (slope bands):           0.05
Maximum number of aspect bands per grid:           1
Minimum cutoff percentage (aspect bands):          0.05
 
#Processor Layout
#Should match the total number of processors used
Number of processors along x:       28
Number of processors along y:       1
 
Halo size along x: 0
Halo size along y: 0 
 
#------------------------ ROUTING -------------------------------------
#HYMAP2 router
Routing model:                "HYMAP2 router"
 
HYMAP2 routing model time step:               "30mn" 
HYMAP2 routing model output interval:         "1da"
HYMAP2 routing model restart interval:        "1mo"
**TEMPLATE model timestep:                      "30mn"**
 
HYMAP2 routing method:                       "local inertia" # "hybrid" #"local inertia" # "kinematic" # 
HYMAP2 routing model time step method:       "adaptive" # "constant" #
HYMAP2 routing model adaptive time step alfa coefficient: .6
HYMAP2 run in ensemble mode:                 0
HYMAP2 routing model linear reservoir flag:  0
HYMAP2 routing model evaporation option:      none # "penman"
HYMAP2 routing model restart file:      ./OUTPUT/BIF_SLR_100_65//ROUTING/200712/LIS_RST_HYMAP2_router_200712312330.d01.nc
HYMAP2 routing model start mode:        restart
 
HYMAP2 floodplain dynamics: 1
 
HYMAP2 reservoir operation option: 0
HYMAP2 number of reservoirs: 1
HYMAP2 reservoir operation input time series length: 900
HYMAP2 reservoir operation input directory: ./volta/
HYMAP2 reservoir operation header filename: ./volta/header_test.txt
HYMAP2 reservoir operation input data type: "water level" # "streamflow" #
 
HYMAP2 routing model dwi flag: 0
 
HYMAP2 enable 2-way coupling: 0
HYMAP2 2-way coupling flooded fraction threshold: .1
 
HYMAP2 discharge direct insertion: 1
HYMAP2 number of gauges: 3
HYMAP2 discharge direct insertion time series length: 210
HYMAP2 discharge direct insertion input directory: ./insertion/200801_200806/
HYMAP2 discharge direct insertion header filename: ./insertion/header.txt
 
HYMAP2 sea level: 1 
HYMAP2 sea level time series length: 210
HYMAP2 sea level input directory: ./sea_level/1993_2022_daily_vlm/200801_200806/
HYMAP2 sea level header filename: ./sea_level/header_sea_level.txt
HYMAP2 sea level outlet list: ./sea_level/outlet_list_002.txt
 
HYMAP2 water management: 0
HYMAP2 water management header file: ./management_rules/miss_management_rules_002.txt
 
HYMAP2 river bifurcation: 1
HYMAP2 river bifurcation pathway file: ./topology_0.bif_subset_1
 
HYMAP2 levee: 0
 
External runoff data source:   "LIS runoff output"
LIS runoff output directory:   ./OUTPUT/test_runs/LAIDA_GVF_100_65/
LIS runoff output interval:    24hr
LIS runoff output domain file: ./lis_input_files/lis_input_delta_coastaldem_002_bif_2hgt_imerg_green_lc_irrig_2008.nc

**Reply by Arijit22IITDelhi** [reply]
Thank you so much; it is working. One more query: is it necessary that the LIS runoff output domain file and the LIS input and parameter data file be the same? like my runoff outputs are at smaller domain than the LIS input file. Though Hymap is running, with this configuration, still I would like to confirm this.

**Comment by agetiran** [comment]
Yes, LIS uses the the file defined in
**LIS runoff output domain file:**
to identify the domain and spatial resolution of your LIS output files.

**Reply by Arijit22IITDelhi** [reply]
Ok, Thank you so much.
